### DISCUSSION TITLE
Coarse/fine boundary update with more than 2 ghost cells

### DISCUSSION THREAD

**Discussion by saisandeepdammati** [PrincetonUniversity/athena#447] [Q&A]
Howdy,

This is Sandeep. I am a Ph.D. student in the department of aerospace engineering at Texas A&M University under Dr. Alexei Poludnenko. We are developing a version of Athena++ with reactive chemistry (both single-step kinetics and multi-step kinetics) for flames and detonations. We have also added additional features into the code such as Visit’s in-situ visualization libsim. 

Currently we are trying to debug an issue with respect to ghost cells. Have you tried running Athena++ with nghost > 2 (for example ghost = 4) with vl2 + plm and AMR enabled. Do you know if all the necessary ghost cells are properly updated between meshblocks  at the coarse/fine boundary when ghost cells are greater than 2. We noticed that all the ghost cells are not updated properly and some of them end up having either zeros or garbage values. Can you point me to the function that decides on how many ghost cells have to be updated between mesh blocks at the coarse/fine boundary?

Thank you very much.

**Comment by felker** [comment]
`NGHOST=4` is guaranteed to be filled correctly in the absence of mesh refinement during the appropriate ghost cell phase exchange phase in the task list. I cannot recall what, if any, guarantees we given on ghost cells beyond the first two layers when AMR is used. @tomidakn ?

**Reply by tomidakn** [reply]
Athena++ (except the Multigrid gravity solver) should fill all the physical variables up to NGHOST even with Adaptive/Static Mesh Refinement, no matter which solver is used. You can see this by setting ghost_zones=1 in an output block (VTK is better for this as you can plot each MeshBlock separately).

Please check the LoadBoundaryBuffer* and SetBoundary* functions in bvals_cc.cpp (for cell centered variables), as well as the MeshBlock constructors in meshblock.cpp to see how the coarse buffer indexes are calculated.

**Comment by c-white** [comment]
@saisandeepdammati Can you clarify which cells seem to not be updated? For example are they in the corners of the block? Which variables are affected?

**Reply by felker** [reply]
the corner ghost cells might not have the values you expect @saisandeepdammati
